pkg/proxy/ipvs/proxier.go

```diff
@@ -1090,15 +850,25 @@ func (proxier *Proxier) syncProxyRules() {
 	}
 
 	// Build IPVS rules for each service.
-	for svcName, svcInfo := range proxier.serviceMap {
+	for svcName, svc := range proxier.serviceMap {
+		svcInfo, ok := svc.(*serviceInfo)
+		if !ok {
+			glog.Errorf("Failed to cast serviceInfo")
+			continue
+		}
 		protocol := strings.ToLower(string(svcInfo.protocol))
 		// Precompute svcNameString; with many services the many calls
 		// to ServicePortName.String() show up in CPU profiles.
 		svcNameString := svcName.String()
 
 		// Handle traffic that loops back to the originator with SNAT.
-		for _, ep := range proxier.endpointsMap[svcName] {
-			epIP := ep.IPPart()
+		for _, e := range proxier.endpointsMap[svcName] {
+			ep, ok := e.(*endpointsInfo)
+			if !ok {
+				glog.Errorf("Failed to cast ep to endpointsInfo")
```

Comment:
Maybe include svcName and e in the log message?

Reply:
Done.